internal/common/resource.go

```diff
@@ -93,8 +96,41 @@ func createOrUpdate(request *Request, resource controllerutil.Object, isClusterR
 
 func setOwner(request *Request, resource controllerutil.Object, isClusterRes bool) error {
 	if isClusterRes {
-		return libhandler.SetOwnerAnnotations(request.Instance, resource)
+		err := libhandler.SetOwnerAnnotations(request.Instance, resource)
+		if err != nil {
+			return err
+		}
+
+		// Removing ownerReferences for objects prior to creation
+		resource.SetOwnerReferences(nil)
+
+		// Removing ownerReferences for existing objects
+		existingRes := newEmptyResource(resource)
+		key, err := client.ObjectKeyFromObject(resource)
+		if err != nil {
+			return err
+		}
+
+		err = request.Client.Get(request.Context, key, existingRes)
+		if err != nil && !errors.IsNotFound(err) {
+			return err
+		} else if err == nil {
+			if len(existingRes.GetOwnerReferences()) > 0 {
+				request.Logger.Info(fmt.Sprintf("Patching %s to remove ownerReferences", key))
+				patch := client.RawPatch(types.JSONPatchType,
+					[]byte(`[{"op": "replace", "path": "/metadata/ownerReferences", "value": []}, {"op": "remove", "path": "/metadata/ownerReferences"}]`))
+				err = request.Client.Patch(request.Context, existingRes, patch)
```

Comment:
Could you explain why this `Patch` call is needed? I thought it would be enough to clear owner references on the resource: `resource.SetOwnerReferences(nil)`.
Then in the `createOrUpdate` function the owner references of the found resource are rewritten.

Reply:
I tried that before, but it did not work, I suspected the `patchStrategy` of `ownerReferences`, I will verify it again

Reply:
Does Update API call use patch strategy?

Reply:
After verifying again, it does not, `SetOwnerReferences(nil)` before an `Update` should be enough

Reply:
I think @vatsalparekh and I experienced similar issue